### DISCUSSION TITLE
Add sample info to 10x VDJ scRNA-seq data

### DISCUSSION THREAD

**Discussion by Celine-loves-coffee** [milaboratory/mixcr#1744] [Q&A] [answered]
Hi Mixcr team,

Thanks for developing the pipeline.

I'm new to coding and have encountered issues with analyzing my BCR data. I have two sequencing samples with a total of 16 groups, and I want to use cell barcodes to identify these groups (information extracted from Seurat).

Here is an example of my sample sheet:
Sample	TagPattern	cell
SP_MT1		AAACCTGAGACCTAGG
SP_MT1		AAACCTGTCGGCGCAT
SP_MT1		AAAGATGTCCCATTAT
SP_MT2		AAAGTAGAGCTGAACG
...

and my command is:
```
mixcr analyze 10x-sc-xcr-vdj \
    -Xmx150g \
    --no-json-reports \
    --force-overwrite \
    --species mmu \
    --sample-sheet Sample_table.tsv \
    /VDJ_raw_data/S{{n}}_L00{{n}}_{{R}}_001.fastq.gz \
    result
```

The error I have is:

The following tags and their roles were recognised before tag transformation:
  Payload tags: R2
  Cell tags: CELL(SQ)
  Molecule tags: UMI(SQ)
  Technical tags: PATTERN_READ_VARIANT_ID(NS)
The following tags and their roles will be associated with each output alignment:
  Payload tags: R2
  Sample tags: Sample(NS)
  Molecule tags: UMI(SQ)
App version: 4.6.0-240-develop; built=Wed Jul 31 07:29:46 GMT+10:00 2024; rev=4adb74a18d; lib=repseqio.v5.0
Must contain at least one Cell tag (determined as tag name starting from "cell" (like "CELL1", "Cell", "CellId", etc..)).

The code runs all good without specifying the sample sheet but fails when including it. Could anyone help me with it? Any suggestion would be much appreciated.

**Comment by mizraelson** [comment]
Hi,
Please use the latest MiXCR v4.7 and also add a "*" symbol to the column name in your Sample_table.tsv file e.g. `CELL*`

**Reply by Celine-loves-coffee** [reply, marked as answer]
Hi mizraelson,

Thanks for the response. I modified the sample sheet file and tried the v4.7.0. 

The updated sample sheet file looks like this:

Sample	TagPattern	CELL*
SP_WT3		CCTAGCTTCAAACAAG
SP_WT3		CTTGGCTAGCAGCCTC
...

and the updated code is:
```
mixcr analyze 10x-sc-xcr-vdj \
    -Xmx150g \
    --no-json-reports \
    --force-overwrite \
    --assemble-contigs-by-cells \
    --sample-sheet updated_sample_table.tsv \
    --species mmu \
    /VDJ_raw_data/S{{n}}_L00{{n}}_{{R}}_001.fastq.gz \
    /result
```
However, I got the following error:

App version: 4.7.0; built=Thu Aug 08 05:19:48 GMT+10:00 2024; rev=976ba14139; lib=repseqio.v5.1
`tagTransformationSteps` are not supported with mitool commands in pipeline

Again, it's good without ``` --sample-sheet updated_sample_table.tsv \```
Any suggestion?

**Reply by mizraelson** [reply]
Sorry about that!

Please try this version:
https://cdn.milaboratory.com/software/mixcr/mixcr-4.6.0-221-keep-tags-from-sample-sheet-16c691bd390f65e3bbba0a452b59254c481be5ed.zip


We will fix this in 4.7.1 soon.

**Reply by mizraelson** [reply]
Hi, just to update this should now work with [the latest develop version](https://link.milaboratory.com/software/mixcr/mixcr-develop.zip) too.
